### ISSUE TITLE
Reserved stock updated even if no payment happened

### ISSUE THREAD

**Issue by cedricfontaine** [PrestaShop/PrestaShop#19100]
Reserved stock is based on order status.

Query seems to be the following :
WHERE
                os.shipped != 1 AND (
                    o.valid = 1 OR (
                        os.id_order_state != :error: AND
                        os.id_order_state != :canceled:
                    )
                )

But we have status related to the payment system : Waiting for payment. It means that the stock won't be available for the customer even if the payment is not valid.

Could you add a new option to manage this particular, to avoid to have stock block in reserved ?

#### Additional information

* PrestaShop version: 1.7.4.0

**Comment by colinegin** [comment]
Hello @cedricfontaine ,

Thanks for your feedback.
So, what you mean is to consider the "waiting for payment" status not a valid status and thus not block the product stock ? 
But what if the payment needs a few days to be validated (take the example of "waiting for wire payment"), it means that there is a risk that the product is OOS when the payment gets valid.
What do you think ?

**Comment by cedricfontaine** [comment]
Yes, there is a risk. But there is also a risk to not sold the product if the payment never go through. We have a case of a product with 69 reserved quantities because of this.
On large website it could be very problematic.

**Comment by colinegin** [comment]
Your improvement idea is interesting but we need to take into account all use cases.
I've added the issue to the Order statuses epic so that we can study it more indepth.

Thanks !

**Comment by Hlavtox** [comment]
This is correct, if you allow other people ordering before the person finishes his payment, you are running yourself into trouble where you accept multiple orders for a single piece of product.
